### PR TITLE
Implementation of spotify_handler

### PR DESCRIPTION
**Pull request by Emmanuel-Melon**
## Description

Initial implementation of the Spotify handler.

**Fixes** #5501 

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)

### What is the solution?

(Describe at a high level how the feature was implemented)

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation, or created issues to update them.
- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/integrations/handlers/spotify_handler/spotify_handler.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from mindsdb.integrations.libs.api_handler import APIHandler, APITable, FuncParser
+
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+
+
+class TwitterHandler(APIHandler):
```
Comment (Smyja) [inline mindsdb/integrations/handlers/spotify_handler/spotify_handler.py:10]: why is this here?

File: mindsdb/integrations/handlers/spotify_handler/spotify_tables.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+from typing import List
+
+from mindsdb.integrations.libs.api_handler import APITable
+from mindsdb.integrations.utilities.sql_utils import extract_comparison_conditions
+from mindsdb.utilities.log import get_log
+
+from mindsdb_sql.parser import ast
+
+logger = get_log("integrations.github_handler")
```
Comment (Smyja) [inline mindsdb/integrations/handlers/spotify_handler/spotify_tables.py:11]: Why are you logging the github handler?

File: mindsdb/integrations/handlers/spotify_handler/spotify_handler.py (hunk 1 above)
Comment (Emmanuel-Melon) [reply mindsdb/integrations/handlers/spotify_handler/spotify_handler.py:10]: Work in progress (WIP). I worked on it last night till 3 AM and had to go to bed. Was trying to grasp the concept of handlers. I've already made updates and will push my local changes shortly.

File: mindsdb/integrations/handlers/spotify_handler/spotify_tables.py (hunk 2 above)
Comment (Emmanuel-Melon) [reply mindsdb/integrations/handlers/spotify_handler/spotify_tables.py:11]: Same

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

**Comment by ZoranPandovski** [comment]
Great progress @Emmanuel-Melon. Once this is ready please sign the CLA so we can review/merge this :)

**Comment by Emmanuel-Melon** [comment]
> Great progress @Emmanuel-Melon. Once this is ready please sign the CLA so we can review/merge this :)

Sure thing!

**Comment by Emmanuel-Melon** [comment]
> I have read the CLA Document and I hereby sign the CLA

I have read the CLA Document and I hereby sign the CLA

**Comment by Emmanuel-Melon** [comment]
> > Great progress @Emmanuel-Melon. Once this is ready please sign the CLA so we can review/merge this :)
> 
> Sure thing! I'd like to make more changes. Should be ready for review tomorrow.

**Comment by ZoranPandovski** [comment]
recheckcla

File: mindsdb/integrations/handlers/spotify_handler/tests/test_spotify_handler.py
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/spotify_handler/tests/test_spotify_handler.py:1]: It would be wonderful if some tests could be implemented as well. I did not see any for the other app handlers. Maybe we can use yours as a baseline to create tests, if you can add them.

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/spotify_handler/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+PARAMETERS = {
+  "repository": "mindsdb/mindsdb",
+  "api_key": "your_api_key",    -- optional Spotify API key
+};
+~~~~
+
+Use the established connection to query your database:
+
+~~~~sql
+SELECT * FROM mindsdb_spotify.issues
```
[the remark this answers is not in the document]
Comment (Emmanuel-Melon) [reply mindsdb/integrations/handlers/spotify_handler/README.md:50]: I had to take a break due to a health issue, and as a result, the PR is still a work in progress. However, I'm grateful to be fully recovered and excited to wrap up this PR as soon as possible.

File: mindsdb/integrations/handlers/spotify_handler/spotify_handler.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+logger = get_log("integrations.spotify_handler")
+
+
+class SpotifHandler(APIHandler):
+    """A class for handling connections and interactions with the Spotify API.
+
+    Attributes:
+        bearer_token (str): The consumer key for the Spotify app.
+        api (tweepy.API): The `tweepy.API` object for interacting with the Spotify API.
```
[the remark this answers is not in the document]
Comment (Emmanuel-Melon) [reply mindsdb/integrations/handlers/spotify_handler/spotify_handler.py:33]: Yes, I was using it as a point of reference.

File: mindsdb/integrations/handlers/spotify_handler/spotify_handler.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            method_name (str): Method name to call (e.g. artits)
+            params (Dict): Params to pass to the API call
+        """
+        if method_name == 'albums':
+            return self._get_albums(params)
+        if method_name == 'artists':
+            return self._get_artists(params)
+        if method_name == 'playlists':
+            return self._get_playlists(params)
+        raise NotImplementedError('Method name {} not supported by Binance API Handler'.format(method_name))
```
[the remark this answers is not in the document]
Comment (Emmanuel-Melon) [reply mindsdb/integrations/handlers/spotify_handler/spotify_handler.py:125]: Will try to resolve these issues as soon as I can. Thanks for the feedback!

File: mindsdb/integrations/handlers/spotify_handler/tests/test_spotify_handler.py
Comment (Emmanuel-Melon) [reply mindsdb/integrations/handlers/spotify_handler/tests/test_spotify_handler.py:1]: That's the plan!

**Comment by ZoranPandovski** [comment]
@Emmanuel-Melon, any updates on this, please?

**Comment by Emmanuel-Melon** [comment]
> @Emmanuel-Melon, any updates on this, please?

I was feeling under the weather, but I'm feeling a lot better now! I'll make sure to finish the PR tomorrow. Sorry about the inconvenience, hope you're doing well!

**Comment by ZoranPandovski** [comment]
@Emmanuel-Melon once you have time feel free to re open this. Thanks
